internal/daemon/daemon.go

```diff
@@ -721,18 +720,19 @@ func (d *Daemon) doReboot(sigCh chan<- os.Signal, waitTimeout time.Duration) err
 	return fmt.Errorf("expected reboot did not happen")
 }
 
-var shutdownMsg = "reboot scheduled to update the system"
+var (
+	syscallReboot = unix.Reboot
+	syscallSync   = unix.Sync
+	timeSleep     = time.Sleep
+)
 
-func rebootImpl(rebootDelay time.Duration) error {
-	if rebootDelay < 0 {
-		rebootDelay = 0
+func rebootImpl(delay time.Duration) error {
+	if delay < 0 {
+		delay = 0
 	}
-	mins := int64(rebootDelay / time.Minute)
-	cmd := exec.Command("shutdown", "-r", fmt.Sprintf("+%d", mins), shutdownMsg)
```

Comment:
This would have been a call to systemd in the past for other systems (/usr/sbin/shutdown -> /bin/systemctl) which do include systemd. Pebble calling this directly under systemd could be bad I would imagine. At the same time I would also imagine having no /sbin/reboot dependency would be perfect for some usescases, maybe even Rocks.

Perhaps we need library offering different reboot implementations, and a Pebble variant specific selection of the reboot implementation. I guess we should try to understand if this is an issue or not for the Pebble use cases.

Reply:
The only problem I see here would be syncing filesystems before reboot. However, I'm concerned about `/sbin/shutdown` having more logic than just calling `sync(2)` and `reboot(2)` that current Pebble users have a dependency on. But I don't think this argument is solid enough to justify the effort of maintaining (and polluting the internal Pebble API with) differing implementations of very trivial functionality.